hles/transformation/src/test/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineValidationSpec.scala

```diff
@@ -0,0 +1,75 @@
+package org.broadinstitute.monster.dap
+
+//import better.files._
+import com.spotify.scio.testing.PipelineSpec
+import org.apache.beam.sdk.options.{PipelineOptions, PipelineOptionsFactory}
+import org.broadinstitute.monster.common.PipelineCoders
+import org.broadinstitute.monster.common.msg.JsonParser
+import org.broadinstitute.monster.dap.PostProcess.errCount
+import org.scalatest
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import better.files.File
+import org.apache.beam.sdk.metrics.MetricName
+
+import scala.io.Source
+
+class HLESurveyTransformationPipelineValidationSpec
+    extends AnyFlatSpec
+    with Matchers
+    with PipelineSpec
+    with PipelineCoders
+    with BeforeAndAfterAll {
+  behavior of "HLESurveyTransformationPipeline"
+
+  val opts: PipelineOptions = PipelineOptionsFactory.fromArgs("--runner=DirectRunner").create()
+
+  // set up a pipeline output directory that we can then clean up
+  val outputDir = File.newTemporaryDirectory()
+  override def afterAll(): Unit = outputDir.delete()
+
+  def pipelineTest(
+    inputPath: String,
+    expectedErrorMessage: String,
+    count: Int = 1
+  ): scalatest.Assertion = {
+    val result = runWithRealContext(opts)(ctx =>
+      HLESurveyTransformationPipelineBuilder.buildPipeline(
+        ctx,
+        Args(
+          this.getClass.getResource(inputPath).toString,
+          outputDir.pathAsString
+        )
+      )
+    ).waitUntilFinish()
```

Comment:
❓ I've been trying to get my head around this with `ScioContext` as well: why `waitUntilFinish` vs `waitUntilDone`?

Reply:
This is replicated from the HCA pipeline validation tests - @raaidbroad can you help us understand the distinction here?

Reply:
`waitUntilDone` Does the following: Wait until the pipeline finishes with the State `DONE` (as opposed to `CANCELLED` or `FAILED`). Throw exception otherwise.

`waitUntilFinish` doesn't throw an exception if the pipeline finishes with a different state. I think using `waitUntilFinish` is a relic of when the approach was to actually throw an exception and break the pipeline; since the approach we take in the hca pipeline is now to log, drop, and carry-on, we don't need to use `waitUntilFinish` to ensure the tests run, sooo might as well use `waitUntilDone` now. Good question.

Reply:
There is a non-zero chance that this also has something to do with running the tests on Dataflow; I ran into some weirdness with tests running locally with the direct runner but then breaking with the dataflow runner. I'ma go check the PR history

Reply:
Don't see any comments/commits that reference dataflow weirdness for that bit. Feel free to test the difference at your own peril 😇

Reply:
Bafflingly, changing it to `waitUntilDone` appears to have broken two unrelated specs: `org.broadinstitute.monster.dap.CslbExtractionPipelineBuilderIntegrationSpec` and `org.broadinstitute.monster.dap.HLESExtractionPipelineBuilderIntegrationSpec`, which failed with this opaque stack trace: ```
[info] org.broadinstitute.monster.dap.HLESExtractionPipelineBuilderIntegrationSpec *** ABORTED ***
[info]   org.apache.beam.sdk.Pipeline$PipelineExecutionException: io.circe.ParsingFailure: expected json value got '<br />...' (line 1, column 1)
[info]   at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:348)
[info]   at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:318)
[info]   at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:213)
[info]   at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:67)
[info]   at org.apache.beam.sdk.Pipeline.run(Pipeline.java:317)
[info]   at org.apache.beam.sdk.Pipeline.run(Pipeline.java:303)
[info]   at com.spotify.scio.ScioContext.execute(ScioContext.scala:609)
[info]   at com.spotify.scio.ScioContext$$anonfun$run$1.apply(ScioContext.scala:597)
[info]   at com.spotify.scio.ScioContext$$anonfun$run$1.apply(ScioContext.scala:585)
[info]   at com.spotify.scio.ScioContext.requireNotClosed(ScioContext.scala:674)
[info]   ...
[info]   Cause: io.circe.ParsingFailure: expected json value got '<br />...' (line 1, column 1)
[info]   ...
```

Reply:
@ThePletch that error is very likely unrelated and due to a 500 on the Redcap server (this happens to me a bunch too, and we need to fix it)

Reply:
Yup I think Drew correctly diagnosed that - I've hit that issue in the past before without making any changes to the code.